collectd_openstack/gnocchi/sender.py

```diff
@@ -124,50 +133,61 @@ def _get_metric_id(self, metername, unit):
             if metric_id is not None:
                 self._meter_ids[metername] = metric_id
             else:
-                self._meter_ids[metername] = self._create_metric(metername, endpoint, unit)
+                self._meter_ids[metername] = self._create_metric(
+                    metername, endpoint, unit)
 
         return self._meter_ids[metername]
 
     def _get_metric(self, metername, endpoint, unit):
         if self.resource_id is None:
             return None
-        url = "{}/v1/resource/generic/{}/metric/".format(endpoint, self.resource_id)
+
+        url = "{}/v1/resource/{}/{}/metric/".format(
```

Comment:
What is the reason for creating a resource of type `cuda` here? We likely already have some sort of convention for creating node metrics due to the low power monitoring. I would imagine that means there already are Gnocchi resources of type `generic` that map to Ironic baremetal node UUIDs. I believe power metrics are planned for all nodes eventually, meaning these resources will exist already in Gnocchi. We can choose to attach these CUDA metrics on to those resources rather than creating a new resource type, preserving the current functionality of having just `generic`. I think this might make it easier in the future when we evaluate whether we want to shift to a more specialized resource type for different uses (i.e. `node` vs `instance` or something.)

My recommendation is to keep this pinned to `generic` - I don't think that would really affect anything else in this changeset.